<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Model-based Analysis of Single Cell Transcriptomics • MAST</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Model-based Analysis of Single Cell Transcriptomics">
<meta property="og:description" content="Methods and models for handling zero-inflated single cell
    assay data.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    

<div id="mast-model-based-analysis-of-single-cell-transcriptomics" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#mast-model-based-analysis-of-single-cell-transcriptomics" class="anchor"></a>MAST: Model-based Analysis of Single-cell Transcriptomics</h1></div>
<p>MAST fits two-part, generalized linear models that are specially adapted for bimodal and/or zero-inflated single cell gene expression data.</p>
<div id="examples-and-vignettes" class="section level2">
<h2 class="hasAnchor">
<a href="#examples-and-vignettes" class="anchor"></a>Examples and vignettes</h2>
<p>MAST supports:</p>
<ul>
<li>Easy importing, subsetting and manipulation of expression matrices</li>
<li>Filtering of low-quality cells</li>
<li>Adaptive thresholding of background noise</li>
<li>Tests for univariate differential expression, with adjustment for covariates</li>
<li>Gene set enrichment analysis, corrected for covariates and gene-gene correlations</li>
<li>Exploration of gene-gene correlations and co-expression</li>
</ul>
<p>Vignettes are available in the package via <code><a href="articles/MAITAnalysis.html">vignette('MAITAnalysis')</a></code> or <code>vignette('MAST-intro')</code>.</p>
</div>
<div id="new-features-and-announcements" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-and-announcements" class="anchor"></a>New Features and announcements</h2>
<ul>
<li>MAST has been ported to use <code>SingleCellExperiment</code> under the hood, and is in <a href="http://bioconductor.org/packages/release/bioc/html/MAST.html">Bioconductor</a>.</li>
<li>We now make an effort to track assay contents (counts vs log counts). This should facilitate interaction with Scater and SCRAN. The older version will remain accessible under branch <em>MASTClassic</em>
</li>
</ul>
</div>
<div id="getting-help" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-help" class="anchor"></a>Getting Help</h2>
<p>For general questions, please submit a question to the <a href="https://support.bioconductor.org/t/MAST/">bioconductor support site</a> so that others can benefit from the discussion.</p>
<p>For bug reports (something seems broken): open a bug report <a href="https://github.com/RGLab/MAST/issues">here</a>.</p>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>If you find MAST useful in your work, please consider citing the paper: <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0844-5">MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data</a> G Finak, A McDavid, M Yajima, J Deng, V Gersuk, AK Shalek, CK Slichter et al Genome biology 16 (1), 278</p>
</div>
<div id="installation-instructions" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-instructions" class="anchor"></a>Installation Instructions</h2>
<p>If you have previously installed the package <code>SingleCellAssay</code> you will want to remove it as <code>MAST</code> supercedes <code>SingleCellAssay</code>. (If both <code>MAST</code> and <code>SingleCellAssay</code> are attached, opaque S4 dispatch errors will result.) Remove it with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"> <span class="kw">remove.packages</span>(<span class="st">'SingleCellAssay'</span>)</a></code></pre></div>
<p>Then you may install or update <code>MAST</code> with:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">install.packages</span>(<span class="st">"BiocManager"</span>) <span class="co"># Needed to install all Bioconductor packages</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">BiocManager<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"MAST"</span>)</a></code></pre></div>
</div>
<div id="converting-old-mastclassic-singlecellassay-objects" class="section level2">
<h2 class="hasAnchor">
<a href="#converting-old-mastclassic-singlecellassay-objects" class="anchor"></a>Converting old MASTClassic SingleCellAssay objects</h2>
<p>If you have data analyzed using MASTClassic, you can convert objects from MASTClassic format to the new format based on SingleCellExperiment using <code>convertMastClassicToSingleCellAssay()</code>.</p>
</div>
</div>

  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/MAST">https://​www.bioconductor.org/​packages/​MAST</a>
</li>
<li>Browse source code at <br><a href="https://github.com/RGLab/MAST/">https://​github.com/​RGLab/​MAST/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/RGLab/MAST/issues">https://​github.com/​RGLab/​MAST/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>GPL(&gt;= 2)</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Andrew McDavid <br><small class="roles"> Author, maintainer </small>  </li>
<li>Greg Finak <br><small class="roles"> Author </small>  </li>
<li>Masanao Yajima <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/RGLab/MAST"><img src="https://travis-ci.org/RGLab/MAST.svg?branch=master" alt="Build Status"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
