<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MAST Intro • MAST</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MAST Intro">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="../articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="../articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>MAST Intro</h1>
                        <h4 class="author">Andrew McDavid</h4>
                        <h4 class="author">Greg Finak</h4>
            
            <h4 class="date">8/23/2017</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RGLab/MAST//blob/master/vignettes/MAST-Intro.Rmd"><code>vignettes/MAST-Intro.Rmd</code></a></small>
      <div class="hidden name"><code>MAST-Intro.Rmd</code></div>

    </div>

    
    
<div id="philosophy" class="section level1">
<h1 class="hasAnchor">
<a href="#philosophy" class="anchor"></a>Philosophy</h1>
<p>MAST is an R/Bioconductor package for managing and analyzing qPCR and sequencing-based single-cell gene expression data, as well as data from other types of single-cell assays. Our goal is to support assays that have multiple <em>features</em> (genes, markers, etc) per <em>well</em> (cell, etc) in a flexible manner. Assays are assumed to be mostly <em>complete</em> in the sense that most <em>wells</em> contain measurements for all features.</p>
<div id="internals" class="section level2">
<h2 class="hasAnchor">
<a href="#internals" class="anchor"></a>Internals</h2>
<p>A <strong>SingleCellAssay</strong> object can be manipulated as a matrix, with rows giving features and columns giving cells. It derives from <a href="**SummarizedExperiment**">http://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html</a>.</p>
</div>
<div id="statistical-testing" class="section level2">
<h2 class="hasAnchor">
<a href="#statistical-testing" class="anchor"></a>Statistical Testing</h2>
<p>Apart from reading and storing single-cell assay data, the package also provides functionality for significance testing of differential expression using a Hurdle model, gene set enrichment, facilities for visualizing patterns in residuals indicative of differential expression, and power calculations (soon).</p>
<p>There is also some facilities for inferring background thresholds, and filtering of individual outlier wells/libraries. These methods are described our papers, <span class="citation">McDavid et al. (2014)</span> , <span class="citation">McDavid et al. (2013)</span> , <span class="citation">Finak et al. (2015)</span> , <span class="citation">McDavid, Finak, and Gottardo (2016)</span> .</p>
</div>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<p>With the cursory background out of the way, we’ll proceed with some examples to help understand how the package is used.</p>
<div id="reading-data" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-data" class="anchor"></a>Reading Data</h2>
<p>Data can be imported in a Fluidigm instrument-specific format (the details of which are undocumented, and likely subject-to-change) or some derived, annotated format, or in “long” (melted) format, in which each row is a measurement, so if there are <span class="math inline">\(N\)</span> wells and <span class="math inline">\(M\)</span> cells, then the <strong>data.frame</strong> should contain <span class="math inline">\(N \times M\)</span> rows.</p>
<p>For example, the following data set was provided in as a comma-separated value file. It has the cycle threshold (<span class="math inline">\(ct\)</span>) recorded. Non-detected genes are recorded as NAs. For the Fluidigm/qPCR single cell expression functions to work as expected, we must use the <em>expression threshold</em>, defined as <span class="math inline">\(et = c_{\mbox{max}} - ct\)</span>, which is proportional to the log-expression.</p>
<p>Below, we load the package and the data, then compute the expression threshold from the <span class="math inline">\(ct\)</span>, and construct a <strong>FluidigmAssay</strong>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">data</span>(vbeta)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">colnames</span>(vbeta)</a></code></pre></div>
<pre><code>##  [1] "Sample.ID"         "Subject.ID"        "Experiment.Number"
##  [4] "Chip.Number"       "Stim.Condition"    "Time"             
##  [7] "Population"        "Number.of.Cells"   "Well"             
## [10] "Gene"              "Ct"</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">vbeta &lt;-<span class="st"> </span><span class="kw"><a href="../reference/computeEtFromCt.html">computeEtFromCt</a></span>(vbeta)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">vbeta.fa &lt;-<span class="st"> </span><span class="kw"><a href="../reference/FromFlatDF.html">FromFlatDF</a></span>(vbeta, <span class="dt">idvars=</span><span class="kw">c</span>(<span class="st">"Subject.ID"</span>, <span class="st">"Chip.Number"</span>, <span class="st">"Well"</span>),</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                          <span class="dt">primerid=</span><span class="st">'Gene'</span>, <span class="dt">measurement=</span><span class="st">'Et'</span>, <span class="dt">ncells=</span><span class="st">'Number.of.Cells'</span>,</a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                          <span class="dt">geneid=</span><span class="st">"Gene"</span>,  <span class="dt">cellvars=</span><span class="kw">c</span>(<span class="st">'Number.of.Cells'</span>, <span class="st">'Population'</span>),</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                          <span class="dt">phenovars=</span><span class="kw">c</span>(<span class="st">'Stim.Condition'</span>,<span class="st">'Time'</span>), <span class="dt">id=</span><span class="st">'vbeta all'</span>, <span class="dt">class=</span><span class="st">'FluidigmAssay'</span>)</a></code></pre></div>
<pre><code>## Assuming data assay in position 1, with name Et is log-transformed.</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">print</span>(vbeta.fa)</a></code></pre></div>
<pre><code>## class: FluidigmAssay 
## dim: 75 456 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(456): Sub01 1 A01 Sub01 1 A02 ... Sub02 3 H10 Sub02 3 H11
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<p>We see that the variable <strong>vbeta</strong> is a <strong>data.frame</strong> from which we construct the <strong>FluidigmAssay</strong> object. The <strong>idvars</strong> is the set of column(s) in <strong>vbeta</strong> that uniquely identify a well (globally), the <strong>primerid</strong> is a column(s) that specify the feature measured at this well. The <strong>measurement</strong> gives the column name containing the log-expression measurement, <strong>ncells</strong> contains the number of cells (or other normalizing factor) for the well. <strong>geneid</strong>, <strong>cellvars</strong>, <strong>phenovars</strong> all specify additional columns to be included in the <strong>featureData</strong>, <strong>phenoData</strong> and <strong>cellData</strong> (). The output is a <strong>FluidigmAssay</strong> object with  wells and  features.</p>
<p>We can access the feature-level metadata and the cell-level metadata using the <strong>mcols</strong> and <strong>colData</strong> accessors.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">mcols</span>(vbeta.fa),<span class="dv">3</span>)</a></code></pre></div>
<pre><code>## DataFrame with 3 rows and 2 columns
##               Gene    primerid
##        &lt;character&gt; &lt;character&gt;
## B3GAT1      B3GAT1      B3GAT1
## BAX            BAX         BAX
## BCL2          BCL2        BCL2</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">head</span>(<span class="kw">colData</span>(vbeta.fa),<span class="dv">3</span>)</a></code></pre></div>
<pre><code>## DataFrame with 3 rows and 9 columns
##             Number.of.Cells            Population    ncells Subject.ID
##                   &lt;integer&gt;           &lt;character&gt; &lt;integer&gt;   &lt;factor&gt;
## Sub01 1 A01               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A02               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A03               1 CD154+VbetaResponsive         1      Sub01
##             Chip.Number        Well Stim.Condition     Time     wellKey
##               &lt;integer&gt; &lt;character&gt;    &lt;character&gt; &lt;factor&gt; &lt;character&gt;
## Sub01 1 A01           1         A01      Stim(SEB)       12 Sub01 1 A01
## Sub01 1 A02           1         A02      Stim(SEB)       12 Sub01 1 A02
## Sub01 1 A03           1         A03      Stim(SEB)       12 Sub01 1 A03</code></pre>
We see this gives us the set of genes measured in the assay, or the cell-level metadata (i.e. the number of cells measured in the well, the population this cell belongs to, the subject it came from, the chip it was run on, the well id, the stimulation it was subjected to, and the timepoint for the experiment this cell was part of). The wellKey are concatenated idvars columns, helping to ensure consistency when splitting and merging SCA objects.

<p>Data can also be imported in matrix format using command <strong>FromMatrix</strong>, and passing a matrix of expression values and <strong>DataFrame</strong> coercible cell and feature data.</p>

<p>It’s possible to subset SingleCellAssay objects by wells and features. Square brackets (“[”) will index on the first index (features) and by features on the second index (cells). Integer and boolean and indices may be used, as well as character vectors naming the wellKey or the feature (via the primerid). There is also a <strong>subset</strong> method, which will evaluate its argument in the frame of the <strong>colData</strong>, hence will subset by wells.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">sub1 &lt;-<span class="st"> </span>vbeta.fa[,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw">show</span>(sub1)</a></code></pre></div>
<pre><code>## class: FluidigmAssay 
## dim: 75 10 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(10): Sub01 1 A01 Sub01 1 A02 ... Sub01 1 A09 Sub01 1 A10
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">sub2 &lt;-<span class="st"> </span><span class="kw">subset</span>(vbeta.fa, Well<span class="op">==</span><span class="st">'A01'</span>)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="kw">show</span>(sub2)</a></code></pre></div>
<pre><code>## class: FluidigmAssay 
## dim: 75 5 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(5): Sub01 1 A01 Sub01 2 A01 Sub02 1 A01 Sub02 2 A01 Sub02
##   3 A01
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">sub3 &lt;-<span class="st"> </span>vbeta.fa[<span class="dv">6</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="kw">show</span>(sub3)</a></code></pre></div>
<pre><code>## class: FluidigmAssay 
## dim: 5 10 
## metadata(0):
## assays(1): Et
## rownames(5): CCL4 CCL5 CCR2 CCR4 CCR5
## rowData names(2): Gene primerid
## colnames(10): Sub01 1 A01 Sub01 1 A02 ... Sub01 1 A09 Sub01 1 A10
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw">colData</span>(sub3)</a></code></pre></div>
<pre><code>## DataFrame with 10 rows and 9 columns
##             Number.of.Cells            Population    ncells Subject.ID
##                   &lt;integer&gt;           &lt;character&gt; &lt;integer&gt;   &lt;factor&gt;
## Sub01 1 A01               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A02               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A03               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A04               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A05               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A06               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A07               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A08               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A09               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A10               1 CD154+VbetaResponsive         1      Sub01
##             Chip.Number        Well Stim.Condition     Time     wellKey
##               &lt;integer&gt; &lt;character&gt;    &lt;character&gt; &lt;factor&gt; &lt;character&gt;
## Sub01 1 A01           1         A01      Stim(SEB)       12 Sub01 1 A01
## Sub01 1 A02           1         A02      Stim(SEB)       12 Sub01 1 A02
## Sub01 1 A03           1         A03      Stim(SEB)       12 Sub01 1 A03
## Sub01 1 A04           1         A04      Stim(SEB)       12 Sub01 1 A04
## Sub01 1 A05           1         A05      Stim(SEB)       12 Sub01 1 A05
## Sub01 1 A06           1         A06      Stim(SEB)       12 Sub01 1 A06
## Sub01 1 A07           1         A07      Stim(SEB)       12 Sub01 1 A07
## Sub01 1 A08           1         A08      Stim(SEB)       12 Sub01 1 A08
## Sub01 1 A09           1         A09      Stim(SEB)       12 Sub01 1 A09
## Sub01 1 A10           1         A10      Stim(SEB)       12 Sub01 1 A10</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">mcols</span>(sub3)</a></code></pre></div>
<pre><code>## DataFrame with 5 rows and 2 columns
##             Gene    primerid
##      &lt;character&gt; &lt;character&gt;
## CCL4        CCL4        CCL4
## CCL5        CCL5        CCL5
## CCR2        CCR2        CCR2
## CCR4        CCR4        CCR4
## CCR5        CCR5        CCR5</code></pre>
<p>The cellData and featureData <strong>AnnotatedDataFrames</strong> are subset accordingly as well.</p>
<p>A SingleCellAssay may be split into a list of SingleCellAssay. The split method takes an argument which names the column (factor) on which to split the data. Each level of the factor will be placed in its own SingleCellAssay within the list.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">sp1 &lt;-<span class="st"> </span><span class="kw">split</span>(vbeta.fa, <span class="st">'Subject.ID'</span>)</a>
<a class="sourceLine" id="cb21-2" data-line-number="2"><span class="kw">show</span>(sp1)</a></code></pre></div>
<pre><code>## $Sub01
## class: FluidigmAssay 
## dim: 75 177 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(177): Sub01 1 A01 Sub01 1 A02 ... Sub01 2 H09 Sub01 2 H10
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):
## 
## $Sub02
## class: FluidigmAssay 
## dim: 75 279 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(279): Sub02 1 A01 Sub02 1 A02 ... Sub02 3 H10 Sub02 3 H11
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<p>The splitting variable can either be a character vector naming column(s) of the SingleCellAssay, or may be a <strong>factor</strong> or <strong>list</strong> of <strong>factor</strong>s.</p>
<p>It’s possible to combine SingleCellAssay objects with the <strong>cbind</strong> method.</p>
<pre><code>## class: FluidigmAssay 
## dim: 75 456 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(456): Sub01 1 A01 Sub01 1 A02 ... Sub02 3 H10 Sub02 3 H11
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
</div>
<div id="filtering" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering" class="anchor"></a>Filtering</h2>
<p>We can filter and perform some significance tests on the SingleCellAssay. We may want to filter any wells with at least two outlier cells where the discrete and continuous parts of the signal are at least 9 standard deviations from the mean. This is a very conservative filtering criteria. We’ll group the filtering by the number of cells.</p>
<p>We’ll split the assay by the number of cells and look at the concordance plot after filtering.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">vbeta.split&lt;-<span class="kw">split</span>(vbeta.fa,<span class="st">"Number.of.Cells"</span>)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="co">#see default parameters for plotSCAConcordance</span></a>
<a class="sourceLine" id="cb24-3" data-line-number="3"><span class="kw"><a href="../reference/plotSCAConcordance.html">plotSCAConcordance</a></span>(vbeta.split[[<span class="dv">1</span>]],vbeta.split[[<span class="dv">2</span>]],</a>
<a class="sourceLine" id="cb24-4" data-line-number="4">                   <span class="dt">filterCriteria=</span><span class="kw">list</span>(<span class="dt">nOutlier =</span> <span class="dv">1</span>, <span class="dt">sigmaContinuous =</span> <span class="dv">9</span>,</a>
<a class="sourceLine" id="cb24-5" data-line-number="5">                       <span class="dt">sigmaProportion =</span> <span class="dv">9</span>))</a></code></pre></div>
<pre><code>## Using primerid as id variables
## Using primerid as id variables
## Using primerid as id variables
## Using primerid as id variables</code></pre>
<pre><code>## Sum of Squares before Filtering: 14.89
##  After filtering: 12.41
##  Difference: 2.48</code></pre>
<p><img src="MAST-Intro_files/figure-html/splitbyncells-1.png" width="384"></p>
<p>The filtering function has several other options, including whether the filter shuld be applied (thus returning a new SingleCellAssay object) or returned as a matrix of boolean values.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1">vbeta.fa</a></code></pre></div>
<pre><code>## class: FluidigmAssay 
## dim: 75 456 
## metadata(0):
## assays(1): Et
## rownames(75): B3GAT1 BAX ... TNFRSF9 TNFSF10
## rowData names(2): Gene primerid
## colnames(456): Sub01 1 A01 Sub01 1 A02 ... Sub02 3 H10 Sub02 3 H11
## colData names(9): Number.of.Cells Population ... Time wellKey
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co">## Split by 'ncells', apply to each component, then recombine</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2">vbeta.filtered &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mast_filter.html">mast_filter</a></span>(vbeta.fa, <span class="dt">groups=</span><span class="st">'ncells'</span>)</a>
<a class="sourceLine" id="cb29-3" data-line-number="3"><span class="co">## Returned as boolean matrix</span></a>
<a class="sourceLine" id="cb29-4" data-line-number="4">was.filtered &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mast_filter.html">mast_filter</a></span>(vbeta.fa, <span class="dt">apply_filter=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb29-5" data-line-number="5"><span class="co">## Wells filtered for being discrete outliers</span></a>
<a class="sourceLine" id="cb29-6" data-line-number="6"><span class="kw">head</span>(<span class="kw">subset</span>(was.filtered, pctout))</a></code></pre></div>
<pre><code>##             intout null pctout
## Sub01 1 D05  FALSE TRUE   TRUE
## Sub01 1 D06  FALSE TRUE   TRUE
## Sub01 1 D07  FALSE TRUE   TRUE
## Sub01 1 D08  FALSE TRUE   TRUE
## Sub01 1 D10  FALSE TRUE   TRUE
## Sub01 1 D11  FALSE TRUE   TRUE</code></pre>
<p>There’s also some functionality for visualizing the filtering.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw"><a href="../reference/mast_filter.html">burdenOfFiltering</a></span>(vbeta.fa, <span class="st">'ncells'</span>, <span class="dt">byGroup=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="MAST-Intro_files/figure-html/burdenOfFiltering-1.png" width="384"></p>
</div>
</div>
<div id="significance-testing-under-the-hurdle-model" class="section level1">
<h1 class="hasAnchor">
<a href="#significance-testing-under-the-hurdle-model" class="anchor"></a>Significance testing under the Hurdle model</h1>
<p>There are two frameworks available in the package. The first framework <strong>zlm</strong> offers a full linear model to allow arbitrary comparisons and adjustment for covariates. The second framework <strong>LRT</strong> can be considered essentially performing t-tests (respecting the discrete/continuous nature of the data) between pairs of groups. <strong>LRT</strong> is subsumed by the first framework, but might be simpler for some users, so has been kept in the package.</p>
<p>We’ll describe <strong>zlm</strong>. Models are specified in terms of the variable used as the measure and covariates present in the <strong>cellData</strong> using symbolic notation, just as the <strong>lm</strong> function in R.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">vbeta<span class="fl">.1</span> &lt;-<span class="st"> </span><span class="kw">subset</span>(vbeta.fa, ncells<span class="op">==</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"><span class="co">## Consider the first 20 genes</span></a>
<a class="sourceLine" id="cb32-3" data-line-number="3">vbeta<span class="fl">.1</span> &lt;-<span class="st"> </span>vbeta<span class="fl">.1</span>[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>,] </a>
<a class="sourceLine" id="cb32-4" data-line-number="4"><span class="kw">head</span>(<span class="kw">colData</span>(vbeta<span class="fl">.1</span>))</a></code></pre></div>
<pre><code>## DataFrame with 6 rows and 9 columns
##             Number.of.Cells            Population    ncells Subject.ID
##                   &lt;integer&gt;           &lt;character&gt; &lt;integer&gt;   &lt;factor&gt;
## Sub01 1 A01               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A02               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A03               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A04               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A05               1 CD154+VbetaResponsive         1      Sub01
## Sub01 1 A06               1 CD154+VbetaResponsive         1      Sub01
##             Chip.Number        Well Stim.Condition     Time     wellKey
##               &lt;integer&gt; &lt;character&gt;    &lt;character&gt; &lt;factor&gt; &lt;character&gt;
## Sub01 1 A01           1         A01      Stim(SEB)       12 Sub01 1 A01
## Sub01 1 A02           1         A02      Stim(SEB)       12 Sub01 1 A02
## Sub01 1 A03           1         A03      Stim(SEB)       12 Sub01 1 A03
## Sub01 1 A04           1         A04      Stim(SEB)       12 Sub01 1 A04
## Sub01 1 A05           1         A05      Stim(SEB)       12 Sub01 1 A05
## Sub01 1 A06           1         A06      Stim(SEB)       12 Sub01 1 A06</code></pre>
<p>Now, for each gene, we can regress on <strong>Et</strong> the factors <strong>Population</strong> and <strong>Subject.ID</strong>.</p>
<p>In each gene, we’ll fit a Hurdle model with a separate intercept for each population and subject. A an S4 object of class “ZlmFit” is returned, containing slots with the genewise coefficients, variance-covariance matrices, etc.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2">zlm.output &lt;-<span class="st"> </span><span class="kw"><a href="../reference/zlm.html">zlm</a></span>(<span class="op">~</span><span class="st"> </span>Population <span class="op">+</span><span class="st"> </span>Subject.ID, vbeta<span class="fl">.1</span>,)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="kw">show</span>(zlm.output)</a></code></pre></div>
<pre><code>## Fitted zlm on 20 genes and 413 cells.
##  Using BayesGLMlike ~ Population + Subject.ID</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1"><span class="co">## returns a data.table with a summary of the fit</span></a>
<a class="sourceLine" id="cb36-2" data-line-number="2">coefAndCI &lt;-<span class="st"> </span><span class="kw">summary</span>(zlm.output, <span class="dt">logFC=</span><span class="ot">FALSE</span>)<span class="op">$</span>datatable</a>
<a class="sourceLine" id="cb36-3" data-line-number="3">coefAndCI &lt;-<span class="st"> </span>coefAndCI[contrast <span class="op">!=</span><span class="st"> '(Intercept)'</span>,]</a>
<a class="sourceLine" id="cb36-4" data-line-number="4">coefAndCI[,contrast<span class="op">:</span><span class="er">=</span><span class="kw">abbreviate</span>(contrast)]</a>
<a class="sourceLine" id="cb36-5" data-line-number="5"><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggplot">ggplot</a></span>(coefAndCI, <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/aes">aes</a></span>(<span class="dt">x=</span>contrast, <span class="dt">y=</span>coef, <span class="dt">ymin=</span>ci.lo, <span class="dt">ymax=</span>ci.hi, <span class="dt">col=</span>component))<span class="op">+</span></a>
<a class="sourceLine" id="cb36-6" data-line-number="6"><span class="st">    </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_linerange">geom_pointrange</a></span>(<span class="dt">position=</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/position_dodge">position_dodge</a></span>(<span class="dt">width=</span>.<span class="dv">5</span>)) <span class="op">+</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/facet_wrap">facet_wrap</a></span>(<span class="op">~</span>primerid) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-7" data-line-number="7"><span class="st">    </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme">theme</a></span>(<span class="dt">axis.text.x=</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/element">element_text</a></span>(<span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">hjust=</span><span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/coord_cartesian">coord_cartesian</a></span>(<span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>, <span class="dv">3</span>))</a></code></pre></div>
<p><img src="MAST-Intro_files/figure-html/zlmExample-1.png" width="576"></p>
<p>Try <strong>?ZlmFit-class</strong> or <strong>showMethods(classes=‘ZlmFit’)</strong> to see a full list of methods. Multicore support is offered by setting <strong>options(mc.cores=4)</strong>, or however many cores your system has.</p>
<p>The combined test for differences in proportion expression/average expression is found by calling a likelihood ratio test on the fitted object. An array of genes, metrics and test types is returned. We’ll plot the -log10 P values by gene and test type.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1">zlm.lr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/lrTest.html">lrTest</a></span>(zlm.output, <span class="st">'Population'</span>)</a></code></pre></div>
<pre><code>## Refitting on reduced model...</code></pre>
<pre><code>## 
## Done!</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="kw">dimnames</span>(zlm.lr)</a></code></pre></div>
<pre><code>## $primerid
##  [1] "B3GAT1" "BAX"    "BCL2"   "CCL2"   "CCL3"   "CCL4"   "CCL5"  
##  [8] "CCR2"   "CCR4"   "CCR5"   "CCR7"   "CD109"  "CD27"   "CD28"  
## [15] "CD38"   "CD3d"   "CD3g"   "CD4"    "CD40LG" "CD45"  
## 
## $test.type
## [1] "cont"   "disc"   "hurdle"
## 
## $metric
## [1] "lambda"     "df"         "Pr(&gt;Chisq)"</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">pvalue &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggplot">ggplot</a></span>(<span class="kw">melt</span>(zlm.lr[,,<span class="st">'Pr(&gt;Chisq)'</span>]), <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/aes">aes</a></span>(<span class="dt">x=</span>primerid, <span class="dt">y=</span><span class="op">-</span><span class="kw">log10</span>(value)))<span class="op">+</span></a>
<a class="sourceLine" id="cb42-2" data-line-number="2"><span class="st">    </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_bar">geom_bar</a></span>(<span class="dt">stat=</span><span class="st">'identity'</span>)<span class="op">+</span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/facet_wrap">facet_wrap</a></span>(<span class="op">~</span>test.type) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/coord_flip">coord_flip</a></span>()</a>
<a class="sourceLine" id="cb42-3" data-line-number="3"><span class="kw">print</span>(pvalue)</a></code></pre></div>
<p><img src="MAST-Intro_files/figure-html/tests-1.png" width="384"></p>
<p>In fact, the <strong>zlm</strong> framework is quite general, and has wrappers for a variety of modeling functions that accept <strong>glm</strong>-like arguments to be used, such as mixed models (using <strong>lme4</strong>). %This is not eval’d because of mysterious errors emanating from lme4 on the bioconductor machines that I am unable to reproduce. % Warning: failed to assign RegisteredNativeSymbol for deepcopy to deepcopy since deepcopy is already defined in the ‘lme4’ namespace % Quitting from lines 230-232 (MAST-intro.Rnw) % Error: processing vignette ‘MAST-intro.Rnw’ failed with diagnostics: % first argument must be a string (of length 1) or native symbol reference % Execution halted</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1"><span class="kw">library</span>(lme4)</a>
<a class="sourceLine" id="cb43-2" data-line-number="2">lmer.output &lt;-<span class="st"> </span><span class="kw"><a href="../reference/zlm.html">zlm</a></span>(<span class="op">~</span><span class="st"> </span>Stim.Condition <span class="op">+</span>(<span class="dv">1</span><span class="op">|</span>Subject.ID), vbeta<span class="fl">.1</span>, <span class="dt">method=</span><span class="st">'glmer'</span>, <span class="dt">ebayes=</span><span class="ot">FALSE</span>)</a></code></pre></div>
<p>By default, we employ Bayesian logistic regression, which imposes a Cauchy prior of the regression coefficients, for the discrete component. This provides reasonable inference under linear separation. We default to regular least squares for the continuous component with an empirical Bayes’ adjustment for the dispersion (variance) estimate. However, the prior can be adjusted (see <strong>defaultPrior</strong>) or eliminated entirely by setting <strong>method=‘glm’</strong> in <strong>zlm</strong>. It is also possible to use Bayesian linear regression for the continuous component by setting <strong>useContinuousBayes=TRUE</strong> in <strong>zlm</strong>. For example:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1"> orig_results &lt;-<span class="st"> </span><span class="kw"><a href="../reference/zlm.html">zlm</a></span>(<span class="op">~</span>Stim.Condition, vbeta<span class="fl">.1</span>)</a>
<a class="sourceLine" id="cb44-2" data-line-number="2"> dp &lt;-<span class="st"> </span><span class="kw"><a href="../reference/defaultPrior.html">defaultPrior</a></span>(<span class="st">'Stim.ConditionUnstim'</span>)</a>
<a class="sourceLine" id="cb44-3" data-line-number="3"> new_results &lt;-<span class="st">  </span><span class="kw"><a href="../reference/zlm.html">zlm</a></span>(<span class="op">~</span>Stim.Condition, vbeta<span class="fl">.1</span>, <span class="dt">useContinuousBayes=</span><span class="ot">TRUE</span>,<span class="dt">coefPrior=</span>dp)</a>
<a class="sourceLine" id="cb44-4" data-line-number="4"> <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/qplot">qplot</a></span>(<span class="dt">x=</span><span class="kw">coef</span>(orig_results, <span class="st">'C'</span>)[, <span class="st">'Stim.ConditionUnstim'</span>],</a>
<a class="sourceLine" id="cb44-5" data-line-number="5">       <span class="dt">y=</span><span class="kw">coef</span>(new_results, <span class="st">'C'</span>)[, <span class="st">'Stim.ConditionUnstim'</span>],</a>
<a class="sourceLine" id="cb44-6" data-line-number="6">       <span class="dt">color=</span><span class="kw">coef</span>(new_results, <span class="st">'D'</span>)[,<span class="st">'(Intercept)'</span>]) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-7" data-line-number="7"><span class="st">     </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/labs">xlab</a></span>(<span class="st">'Default prior'</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/labs">ylab</a></span>(<span class="st">'Informative Prior'</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb44-8" data-line-number="8"><span class="st">     </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/geom_abline">geom_abline</a></span>(<span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">lty=</span><span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/scale_gradient">scale_color_continuous</a></span>(<span class="st">'Baseline</span><span class="ch">\n</span><span class="st">log odds</span><span class="ch">\n</span><span class="st">of expression'</span>)</a></code></pre></div>
<p><img src="MAST-Intro_files/figure-html/advancedArmExample-1.png" width="384"></p>
<p>After applying a prior to the continuous component, its estimates are shrunken towards zero, with the amount of shrinkage inversely depending on the number of expressing cells in the gene.</p>
<div id="two-sample-likelihood-ratio-test" class="section level2">
<h2 class="hasAnchor">
<a href="#two-sample-likelihood-ratio-test" class="anchor"></a>Two-sample Likelihood Ratio Test</h2>
<p>Another way to test for differential expression is available through the <strong>LRT</strong> function, which is analogous to two-sample T tests.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">two.sample &lt;-<span class="st"> </span><span class="kw"><a href="../reference/LRT.html">LRT</a></span>(vbeta<span class="fl">.1</span>, <span class="st">'Population'</span>, <span class="dt">referent=</span><span class="st">'CD154+VbetaResponsive'</span>)</a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="kw">head</span>(two.sample) </a></code></pre></div>
<pre><code>##              Population test.type primerid    lrstat direction   p.value
## 1 CD154-VbetaResponsive      comb   B3GAT1 1.5722257        -1 0.4556124
## 2 CD154-VbetaResponsive      comb      BAX 1.5847859         1 0.4527601
## 3 CD154-VbetaResponsive      comb     BCL2 0.7821003        -1 0.6763462
## 4 CD154-VbetaResponsive      comb     CCL2 3.3431031        -1 0.1879552
## 5 CD154-VbetaResponsive      comb     CCL3 0.1401862        -1 0.9323070
## 6 CD154-VbetaResponsive      comb     CCL4 0.6305951         1 0.7295718</code></pre>
<p>Here we compare each population (<strong>CD154-VbetaResponsive, CD154-VbetaUnresponsive CD154+VbetaUnresponsive, VbetaResponsive, VbetaUnresponsive</strong>) to the <strong>CD154+VbetaResponsive</strong> population. The <strong>Population</strong> column shows which population is being compared, while <strong>test.type</strong> is <strong>comb</strong> for the combined normal theory/binomial test. Column <strong>primerid</strong> gives the gene being tested, <strong>direction</strong> shows if the comparison group mean is greater (1) or less (-1) than the referent group, and <strong>lrtstat</strong> and <strong>p.value</strong> give the test statistic and <span class="math inline">\(\chi^2\)</span> p-value (two degrees of freedom). Other options are whether additional information about the tests are returned (<strong>returnall=TRUE</strong>) and if the testing should be stratified by a character vector naming columns in <strong>colData</strong> containing grouping variables (<strong>groups</strong>).</p>
<p>These tests have been subsumed by <strong>zlm</strong> but remain in the package for user convenience.</p>
</div>
</div>
<div id="use-with-single-cell-rna-sequencing-data" class="section level1">
<h1 class="hasAnchor">
<a href="#use-with-single-cell-rna-sequencing-data" class="anchor"></a>Use with single cell RNA-sequencing data</h1>
<p>In RNA-sequencing data is essentially no different than qPCR-based single cell gene expression, once it has been aligned and mapped, if one is willing to reduce the experiment to counts or count-like data for a fixed set of genes/features.<br>
We assume that suitable tools (eg, RSEM, Kallisto or TopHat) have been applied to do this.</p>
<p>An example of this use is provided in a vignette. Type <strong>vignette(‘MAITAnalysis’)</strong> to view.</p>
<div id="a-comment-on-implementation-details" class="section level3">
<h3 class="hasAnchor">
<a href="#a-comment-on-implementation-details" class="anchor"></a>A Comment on Implementation Details</h3>
<p>Here we provide some background on the implementation of the package.</p>
<p>There are several fundamental new object types provided by the package. <strong>SummarizedExperiment</strong> is the base class, which is provides an array-like object to store tabular data that might have multiple derived representations. On construction of a <strong>SingleCellAssay</strong> object, the package tests for completeness, and will fill in the missing data (with NA) if it is not, so assays with lots of missing data can make reading marginally slower.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Finak2015-uz">
<p>Finak, Greg, Andrew McDavid, Masanao Yajima, Jingyuan Deng, Vivian Gersuk, Alex K Shalek, Chloe K Slichter, et al. 2015. “MAST: A Flexible Statistical Framework for Assessing Transcriptional Changes and Characterizing Heterogeneity in Single-Cell RNA Sequencing Data.” <em>Genome Biol.</em> 16 (1): 1–13.</p>
</div>
<div id="ref-McDavid2014-kr">
<p>McDavid, Andrew, Lucas Dennis, Patrick Danaher, Greg Finak, Michael Krouse, Alice Wang, Philippa Webster, Joseph Beechem, and Raphael Gottardo. 2014. “Modeling Bi-Modality Improves Characterization of Cell Cycle on Gene Expression in Single Cells.” Edited by Sheng Zhong. <em>PLoS Comput. Biol.</em> 10 (7): e1003696.</p>
</div>
<div id="ref-McDavid2013-mc">
<p>McDavid, Andrew, Greg Finak, Pratip K Chattopadyay, Maria Dominguez, Laurie Lamoreaux, Steven S Ma, Mario Roederer, and Raphael Gottardo. 2013. “Data Exploration, Quality Control and Testing in Single-Cell qPCR-based Gene Expression Experiments.” <em>Bioinformatics</em> 29 (4): 461–67.</p>
</div>
<div id="ref-McDavid2016-rm">
<p>McDavid, Andrew, Greg Finak, and Raphael Gottardo. 2016. “The Contribution of Cell Cycle to Heterogeneity in Single-Cell RNA-seq Data.” <em>Nat. Biotechnol.</em> 34 (6): 591–93.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#philosophy">Philosophy</a><ul class="nav nav-pills nav-stacked">
<li><a href="#internals">Internals</a></li>
      <li><a href="#statistical-testing">Statistical Testing</a></li>
      </ul>
</li>
      <li>
<a href="#examples">Examples</a><ul class="nav nav-pills nav-stacked">
<li><a href="#reading-data">Reading Data</a></li>
      <li><a href="#filtering">Filtering</a></li>
      </ul>
</li>
      <li>
<a href="#significance-testing-under-the-hurdle-model">Significance testing under the Hurdle model</a><ul class="nav nav-pills nav-stacked">
<li><a href="#two-sample-likelihood-ratio-test">Two-sample Likelihood Ratio Test</a></li>
      </ul>
</li>
      <li><a href="#use-with-single-cell-rna-sequencing-data">Use with single cell RNA-sequencing data</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
