<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interoptability between MAST and SingleCellExperiment-derived packages. • MAST</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Interoptability between MAST and SingleCellExperiment-derived packages.">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="../articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="../articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Interoptability between MAST and SingleCellExperiment-derived packages.</h1>
                        <h4 class="author">Andrew McDavid</h4>
            
            <h4 class="date">2018-10-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/RGLab/MAST//blob/master/vignettes/MAST-interoperability.Rmd"><code>vignettes/MAST-interoperability.Rmd</code></a></small>
      <div class="hidden name"><code>MAST-interoperability.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>As a SingleCellExperiment-derived package, <code>MAST</code> can easily be inserted into workflows with packages such as <code>scran</code>, <code>scater</code>, <code>zinbwave</code>, <code>SCnorm</code> and others. The main gotcha is packages that assume integer counts vs log-transformed, or log-transformed, approximately scale-normalized data. We find that MAST performs best with log-transformed, scale-normalized data that has been thresholded, such as <span class="math inline">\(\log_2(\text{transcripts per million} + 1)\)</span>.</p>
<p>We address this by:</p>
<ul>
<li>testing for log-like data for objects constructed in MAST</li>
<li>explicitly naming the slot of the <code>assay</code> containing such putatively log-like data</li>
<li>by default operating on the slot with such log-like data</li>
</ul>
<p>In objects that were constructed in other packages, we …</p>
<p>In what follows, we show an example of using <code>scater</code> to plot some QC metrics, <code>SCnorm</code> to normalize data, and, and conversion to a <code>Seurat</code> object.</p>
</div>
<div id="from-mast-to-scater" class="section level1">
<h1 class="hasAnchor">
<a href="#from-mast-to-scater" class="anchor"></a>From MAST to Scater</h1>
<p>Scater (citation) is a package that …</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(MAST)</a></code></pre></div>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colMeans, colnames, colSums, dirname, do.call, duplicated,
##     eval, evalq, Filter, Find, get, grep, grepl, intersect,
##     is.unsorted, lapply, Map, mapply, match, mget, order, paste,
##     pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply</code></pre>
<pre><code>## Registered S3 methods overwritten by 'ggplot2':
##   method         from 
##   [.quosures     rlang
##   c.quosures     rlang
##   print.quosures rlang</code></pre>
<pre><code>## Registered S3 method overwritten by 'dplyr':
##   method               from  
##   as.data.frame.tbl_df tibble</code></pre>
<pre><code>## 
## Attaching package: 'MAST'</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     filter</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">message =</span> <span class="ot">FALSE</span>,<span class="dt">error =</span> <span class="ot">FALSE</span>,<span class="dt">warning =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="kw">data</span>(maits, <span class="dt">package=</span><span class="st">'MAST'</span>)</a>
<a class="sourceLine" id="cb31-3" data-line-number="3">unlog &lt;-<span class="st"> </span><span class="cf">function</span>(x) <span class="kw">ceiling</span>(<span class="dv">2</span><span class="op">^</span>x <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4">sca_raw =<span class="st">  </span><span class="kw"><a href="../reference/FromMatrix.html">FromMatrix</a></span>(<span class="kw">t</span>(maits<span class="op">$</span>expressionmat), maits<span class="op">$</span>cdat, maits<span class="op">$</span>fdat)</a></code></pre></div>
<pre><code>## Assuming data assay in position 1, with name et is log-transformed.</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw">assays</span>(sca_raw)<span class="op">$</span>counts =<span class="st"> </span><span class="kw">unlog</span>(<span class="kw">assay</span>(sca_raw))</a>
<a class="sourceLine" id="cb33-2" data-line-number="2"><span class="kw">assayNames</span>(sca_raw)</a></code></pre></div>
<p>Here we make an object with assays <code>counts</code> and <code>et</code>. By default, <code>MAST</code> will operate on the <code>et</code> assay, but scran wants count-like data for some of its QC. The <code>et</code> data are log2 + 1 transcripts per million (TPM), as output by RSEM.</p>
<p>We could specify the assay name at creation with <code>sca_raw = FromMatrix(list(logTPM = t(maits$expressionmat)), maits$cdat, maits$fdat)</code> or rename an object that contains appropriately transformed data with <code>assayNames(sca_raw) = c('logTPM', 'counts')</code>.</p>
<p>Before calling <code>scater</code> functionality, you might pause to consider if some features should belong in special <code>control</code> sets, such as mitochrondial genes, or spike-ins.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">library</span>(scater)</a>
<a class="sourceLine" id="cb34-2" data-line-number="2">sca_raw =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/calculateQCMetrics">calculateQCMetrics</a></span>(sca_raw)</a>
<a class="sourceLine" id="cb34-3" data-line-number="3"><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plotRowData">plotRowData</a></span>(sca_raw, <span class="dt">x =</span> <span class="st">'log10_mean_counts'</span>, <span class="st">'pct_dropout_by_counts'</span>)</a></code></pre></div>
<p><img src="MAST-interoperability_files/figure-html/scaterQC-1.png" width="700"></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plotColData">plotColData</a></span>(sca_raw, <span class="dt">y=</span><span class="st">"total_features_by_counts"</span>, <span class="dt">x=</span><span class="st">"total_counts"</span>)</a></code></pre></div>
<p><img src="MAST-interoperability_files/figure-html/scaterQC-2.png" width="700"></p>
<p>Evidently some features were filtered, so not all cells contain 1 million counts. We can tell these were rare features based on the inverse relationship between <code>total_counts</code> and <code>total_features_by_counts</code>: the most complex libraries (with the greatest numer of features) are missing the most counts.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">sca_raw &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sca_raw, <span class="dt">ncomponents=</span><span class="dv">5</span>, <span class="dt">exprs_values =</span> <span class="st">'et'</span>)</a>
<a class="sourceLine" id="cb36-2" data-line-number="2"><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plotReducedDim">plotReducedDim</a></span>(sca_raw, <span class="dt">use_dimred =</span> <span class="st">'PCA'</span>, <span class="dt">colour_by =</span> <span class="st">'condition'</span>)</a></code></pre></div>
<p><img src="MAST-interoperability_files/figure-html/unnamed-chunk-2-1.png" width="700"> We can also run a PCA.</p>
<div id="from-scater-to-mast" class="section level2">
<h2 class="hasAnchor">
<a href="#from-scater-to-mast" class="anchor"></a>From scater to MAST</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="kw">data</span>(sc_example_counts)</a></code></pre></div>
</div>
</div>
<div id="wip-from-mast-to-zinb-wave" class="section level1">
<h1 class="hasAnchor">
<a href="#wip-from-mast-to-zinb-wave" class="anchor"></a>WIP: From MAST to ZINB-wave</h1>
<!-- ```{r zinbwave} -->
<!-- library(zinbwave) -->
<!-- feature_var = apply(assay(sca_raw), 1, var) -->
<!-- sca_top500 = sca_raw[rank(-feature_var)<=500,] -->
<!-- zw <- zinbwave(Y = sca_top500, X = '~condition', which_assay = 'counts') -->
<!-- ``` -->
<!-- Run zinbwave.  To speed things, we take the -->
<!-- top 500 most variable genes. -->
<!-- ```{r} -->
<!-- ggpairs(data.frame(zw = reducedDim(zw), colData(zw)), columns = c('zw.PC1', 'zw.PC2', 'ngeneson'), mapping = aes(color = condition)) -->
<!-- ``` -->
<!-- ## Using MAST to characterizing genes that drive the factors -->
<!-- ```{r, results = 'hide'} -->
<!-- assays(zw)[[1]] <- log2(assay(zw)+1) -->
<!-- colData(zw) <- cbind(colData(zw), zw=as.data.frame(reducedDim(zw))) -->
<!-- zw <- zw %>% as('SingleCellAssay') -->
<!-- zz <- zlm(~W1+W2, sca=zw) -->
<!-- ``` -->
<!-- ```{r, results = 'asis'} -->
<!-- ss <- summary(zz) -->
<!-- knitr::kable(print(ss)) -->
<!-- ``` -->
<!-- These are log-fold changes in the top few changes associated with factors 1 and 2. -->
<!-- ```{r} -->
<!-- top5 <- ss$datatable %>% filter(component=='logFC', contrast %like% 'W') %>% arrange(-abs(z)) %>% head(n=5) %>% left_join(rowData(zw) %>% as.data.table()) -->
<!-- plt <- ggplot(zw[top5$primerid,] %>% as('data.table'), aes(x=W1, color = condition)) + geom_point() + facet_wrap(~symbolid) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- plt + aes(y = et) -->
<!-- ``` -->
<!-- Expression on "Et" scale ($\log_2( TPM + 1)$) -->
<!-- ```{r} -->
<!-- plt + aes(y = normalizedValues) -->
<!-- ``` -->
<!-- Normalized expression from factor model -->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#from-mast-to-scater">From MAST to Scater</a><ul class="nav nav-pills nav-stacked">
<li><a href="#from-scater-to-mast">From scater to MAST</a></li>
      </ul>
</li>
      <li><a href="#wip-from-mast-to-zinb-wave">WIP: From MAST to ZINB-wave</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
