<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get the concordance between two experiments — getConcordance • MAST</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Get the concordance between two experiments — getConcordance" />

<meta property="og:description" content="Return the concordance between two assays (i.e. single cell and hundred cell).
The &quot;average&quot; of singleCellRef (after adjusting for the number of cells) and
singleCellComp are taken per gene, per groups.
A data.frame with one row per gene-groups is returned with some additional columns." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="../articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="../articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get the concordance between two experiments</h1>
    <small class="dont-index">Source: <a href='https://github.com/RGLab/MAST//blob/master/R/Fluidigm-methods.R'><code>R/Fluidigm-methods.R</code></a></small>
    <div class="hidden name"><code>getConcordance.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Return the concordance between two assays (i.e. single cell and hundred cell).
The "average" of <code>singleCellRef</code> (after adjusting for the number of cells) and
<code>singleCellComp</code> are taken per gene, per <code>groups</code>.
A <code>data.frame</code> with one row per gene-<code>groups</code> is returned with some additional columns.</p>
    
    </div>

    <pre class="usage"><span class='fu'>getConcordance</span>(<span class='no'>singleCellRef</span>, <span class='no'>singleCellcomp</span>, <span class='kw'>groups</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>fun.natural</span> <span class='kw'>=</span> <span class='no'>expavg</span>, <span class='kw'>fun.cycle</span> <span class='kw'>=</span> <span class='no'>logmean</span>)

<span class='fu'>getwss</span>(<span class='no'>concord</span>, <span class='no'>nexp</span>)

<span class='fu'>getss</span>(<span class='no'>concord</span>)

<span class='fu'>getrc</span>(<span class='no'>concord</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>singleCellRef</th>
      <td><p>"reference" SingleCellAssay</p></td>
    </tr>
    <tr>
      <th>singleCellcomp</th>
      <td><p>"comparison" SingleCellAssay</p></td>
    </tr>
    <tr>
      <th>groups</th>
      <td><p>character vector giving variable(s) on which the comparison is conditioned</p></td>
    </tr>
    <tr>
      <th>fun.natural</th>
      <td><p>function to transform the SingleCellAssays to a mRNA proportional level</p></td>
    </tr>
    <tr>
      <th>fun.cycle</th>
      <td><p>inverse function of fun.natural</p></td>
    </tr>
    <tr>
      <th>concord</th>
      <td><p>data.frame returned by getConcordance</p></td>
    </tr>
    <tr>
      <th>nexp</th>
      <td><p>number of expressed cells per row in <code>concord</code></p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>concordance between two assays</p>
    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>getwss</code>: getrc the sum of squares, weighted by nexp</p></li>
<li><p><code>getss</code>: return the sum of squares</p></li>
<li><p><code>getrc</code>: Return Lin's (1989) concordance correlation coefficient</p></li>
</ul>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='plotSCAConcordance.html'>plotSCAConcordance</a></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>vbetaFA</span>)
<span class='no'>sca1</span> <span class='kw'>&lt;-</span> <span class='fu'>subset</span>(<span class='no'>vbetaFA</span>, <span class='no'>ncells</span><span class='kw'>==</span><span class='fl'>1</span>)
<span class='no'>sca100</span> <span class='kw'>&lt;-</span> <span class='fu'>subset</span>(<span class='no'>vbetaFA</span>, <span class='no'>ncells</span><span class='kw'>==</span><span class='fl'>100</span>)
<span class='no'>concord</span> <span class='kw'>&lt;-</span> <span class='fu'>getConcordance</span>(<span class='no'>sca1</span>, <span class='no'>sca100</span>)</div><div class='output co'>#&gt; <span class='message'>Using primerid as id variables</span></div><div class='output co'>#&gt; <span class='message'>Using primerid as id variables</span></div><div class='input'><span class='fu'>getss</span>(<span class='no'>concord</span>)</div><div class='output co'>#&gt; [1] 3.621746</div><div class='input'><span class='fu'>getrc</span>(<span class='no'>concord</span>)</div><div class='output co'>#&gt; [1] 0.8374755</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#functions">Functions</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Andrew McDavid

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

