<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ MAST</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="../articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="../articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-loading-transforming-and-summarizing" class="hasAnchor"><a href="#section-loading-transforming-and-summarizing" class="anchor"></a>Loading, transforming and summarizing</h2>
          <p class="section-desc"><p>Functions to load data, add covariates or annotations, or provide high-level summaries.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="FromFlatDF.html">FromFlatDF()</a></code> <code><a href="FromFlatDF.html">FluidigmAssay()</a></code> </p>
        </td>
        <td><p>Construct a SingleCellAssay (or derived subclass) from a `flat` (melted) data.frame/data.table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FromMatrix.html">FromMatrix()</a></code> </p>
        </td>
        <td><p>Construct a SingleCellAssay from a matrix or array of expression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="SceToSingleCellAssay.html">SceToSingleCellAssay()</a></code> </p>
        </td>
        <td><p>Coerce a SingleCellExperiment to some class defined in MAST</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convertMASTClassicToSingleCellAssay.html">convertMASTClassicToSingleCellAssay()</a></code> </p>
        </td>
        <td><p>Convert a MASTClassic SingleCellAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="defaultAssay.html">magic_assay_names()</a></code> <code><a href="defaultAssay.html">assay(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> </p>
        </td>
        <td><p>Default <code>assay</code> returned</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="melt.SingleCellAssay.html">melt(<i>&lt;SingleCellAssay&gt;</i>)</a></code> </p>
        </td>
        <td><p>"Melt" a <code>SingleCellAssay</code> matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="zlm.html">zlm()</a></code> </p>
        </td>
        <td><p>Zero-inflated regression for SingleCellAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="freq.html">freq()</a></code> <code><a href="freq.html">condmean()</a></code> <code><a href="freq.html">condSd()</a></code> <code><a href="freq.html">numexp()</a></code> </p>
        </td>
        <td><p>Summary statistics for genes in an experiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getConcordance.html">getConcordance()</a></code> <code><a href="getConcordance.html">getwss()</a></code> <code><a href="getConcordance.html">getss()</a></code> <code><a href="getConcordance.html">getrc()</a></code> </p>
        </td>
        <td><p>Get the concordance between two experiments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.fluidigm.html">read.fluidigm()</a></code> </p>
        </td>
        <td><p>Reads a Fluidigm Biomark (c. 2011) raw data file (or set of files)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="primerAverage.html">primerAverage()</a></code> </p>
        </td>
        <td><p>Average expression values for duplicated/redundant genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mast_filter.html">mast_filter()</a></code> <code><a href="mast_filter.html">burdenOfFiltering()</a></code> </p>
        </td>
        <td><p>Filter a SingleCellAssay</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-thresholding" class="hasAnchor"><a href="#section-thresholding" class="anchor"></a>Thresholding</h2>
          <p class="section-desc"><p>Adaptive thresholding of background noise.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="thresholdSCRNACountMatrix.html">thresholdSCRNACountMatrix()</a></code> </p>
        </td>
        <td><p>Threshold a count matrix using an adaptive threshold.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary.thresholdSCRNACountMatrix.html">summary(<i>&lt;thresholdSCRNACountMatrix&gt;</i>)</a></code> <code><a href="summary.thresholdSCRNACountMatrix.html">print(<i>&lt;summaryThresholdSCRNA&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize the effect of thresholding</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.thresholdSCRNACountMatrix.html">plot(<i>&lt;thresholdSCRNACountMatrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot cutpoints and densities for thresholding</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-differential-expression-testing-with-hurdle-model" class="hasAnchor"><a href="#section-differential-expression-testing-with-hurdle-model" class="anchor"></a>Differential Expression Testing with Hurdle Model</h2>
          <p class="section-desc"><p>Fit a Hurdle linear model to test for zero-inflated differential expression.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="zlm.html">zlm()</a></code> </p>
        </td>
        <td><p>Zero-inflated regression for SingleCellAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="LMERlike-class.html">update(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">vcov(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">coef(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">logLik(<i>&lt;LMERlike&gt;</i>)</a></code> </p>
        </td>
        <td><p>Wrapper for lmer/glmer</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="BayesGLMlike-class.html">BayesGLMlike-class</a></code> </p>
        </td>
        <td><p>Wrapper for bayesian GLM</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.summaryZlmFit.html">print(<i>&lt;summaryZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>Print summary of a ZlmFit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ebayes.html">ebayes()</a></code> </p>
        </td>
        <td><p>Estimate hyperparameters for hierarchical variance model for continuous component</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="defaultPrior.html">defaultPrior()</a></code> </p>
        </td>
        <td><p>Initialize a prior to be used a prior for BayeGLMlike/BayesGLMlike2</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-specifying-hypothesis-and-contrasts-to-test" class="hasAnchor"><a href="#section-specifying-hypothesis-and-contrasts-to-test" class="anchor"></a>Specifying hypothesis and contrasts to test</h2>
          <p class="section-desc"><p>Testing linear functions of coefficients.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="Hypothesis.html">Hypothesis()</a></code> </p>
        </td>
        <td><p>Describe a linear model hypothesis to be tested</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lrTest.html">lrTest()</a></code> </p>
        </td>
        <td><p>Run a likelihood-ratio test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="waldTest.html">waldTest()</a></code> </p>
        </td>
        <td><p>Run a Wald test</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-low-level-manipulation-of-fitted-hurdle-models" class="hasAnchor"><a href="#section-low-level-manipulation-of-fitted-hurdle-models" class="anchor"></a>Low-level manipulation of fitted hurdle models</h2>
          <p class="section-desc"><p>Return coefficients, standard errors, etc. Calculate and return residuals from models.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="BayesGLMlike-class.html">BayesGLMlike-class</a></code> </p>
        </td>
        <td><p>Wrapper for bayesian GLM</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CovFromBoots.html">CovFromBoots()</a></code> </p>
        </td>
        <td><p>Extract the intere-gene covariance matrices for continuous and discrete components of a MAST model for a given coefficient from bootstrap replicates</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Drop.html">Drop()</a></code> </p>
        </td>
        <td><p>Drop specified dimension from an array</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FromFlatDF.html">FromFlatDF()</a></code> <code><a href="FromFlatDF.html">FluidigmAssay()</a></code> </p>
        </td>
        <td><p>Construct a SingleCellAssay (or derived subclass) from a `flat` (melted) data.frame/data.table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="FromMatrix.html">FromMatrix()</a></code> </p>
        </td>
        <td><p>Construct a SingleCellAssay from a matrix or array of expression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GLMlike-class.html">vcov(<i>&lt;GLMlike&gt;</i>)</a></code> </p>
        </td>
        <td><p>Wrapper for regular glm/lm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="GSEATests-class.html">GSEATests-class</a></code> </p>
        </td>
        <td><p>An S4 class for Gene Set Enrichment output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="Hypothesis.html">Hypothesis()</a></code> </p>
        </td>
        <td><p>Describe a linear model hypothesis to be tested</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="LMERlike-class.html">update(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">vcov(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">coef(<i>&lt;LMERlike&gt;</i>)</a></code> <code><a href="LMERlike-class.html">logLik(<i>&lt;LMERlike&gt;</i>)</a></code> </p>
        </td>
        <td><p>Wrapper for lmer/glmer</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="LMlike-class.html">summary(<i>&lt;LMlike&gt;</i>)</a></code> <code><a href="LMlike-class.html">update(<i>&lt;LMlike&gt;</i>)</a></code> <code><a href="LMlike-class.html">waldTest(<i>&lt;LMlike&gt;</i>,<i>&lt;CoefficientHypothesis&gt;</i>)</a></code> <code><a href="LMlike-class.html">waldTest(<i>&lt;LMlike&gt;</i>,<i>&lt;matrix&gt;</i>)</a></code> <code><a href="LMlike-class.html">lrTest(<i>&lt;LMlike&gt;</i>,<i>&lt;character&gt;</i>)</a></code> <code><a href="LMlike-class.html">lrTest(<i>&lt;LMlike&gt;</i>,<i>&lt;CoefficientHypothesis&gt;</i>)</a></code> <code><a href="LMlike-class.html">lrTest(<i>&lt;LMlike&gt;</i>,<i>&lt;Hypothesis&gt;</i>)</a></code> <code><a href="LMlike-class.html">lrTest(<i>&lt;LMlike&gt;</i>,<i>&lt;matrix&gt;</i>)</a></code> <code><a href="LMlike-class.html">logLik(<i>&lt;GLMlike&gt;</i>)</a></code> </p>
        </td>
        <td><p>Linear Model-like Class</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="LRT.html">LRT()</a></code> </p>
        </td>
        <td><p>Likelihood Ratio Tests for SingleCellAssays</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAST-package.html">MAST-package</a></code> </p>
        </td>
        <td><p>MAST: Model-based Analysis of Single- cell Transcriptomics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="SceToSingleCellAssay.html">SceToSingleCellAssay()</a></code> </p>
        </td>
        <td><p>Coerce a SingleCellExperiment to some class defined in MAST</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ZlmFit-class.html">lrTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;CoefficientHypothesis&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">lrTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;Hypothesis&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">lrTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;matrix&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">waldTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;CoefficientHypothesis&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">waldTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;Hypothesis&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">coef(<i>&lt;ZlmFit&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">vcov(<i>&lt;ZlmFit&gt;</i>)</a></code> <code><a href="ZlmFit-class.html">se.coef(<i>&lt;ZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>An S4 class to hold the output of a call to zlm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="applyFlat.html">applyFlat()</a></code> </p>
        </td>
        <td><p>Apply a vectorized binary operation recycling over last dimension</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bootVcov1.html">bootVcov1()</a></code> </p>
        </td>
        <td><p>Bootstrap a zlmfit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cData.html">cData()</a></code> <code><a href="cData.html">`cData<-`()</a></code> <code><a href="cData.html">combine(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;SingleCellAssay&gt;</i>)</a></code> <code><a href="cData.html">combine(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> </p>
        </td>
        <td><p>Deprecated cell/feature data accessors/mutators</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calcZ.html">calcZ()</a></code> </p>
        </td>
        <td><p>Get Z or T statistics and P values after running gseaAfterBoot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="colData-set-SingleCellAssay-DataFrame-method.html">`colData<-`(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;DataFrame&gt;</i>)</a></code> </p>
        </td>
        <td><p>Replace <code>colData</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="collectResiduals.html">collectResiduals()</a></code> <code><a href="collectResiduals.html">discrete_residuals_hook()</a></code> <code><a href="collectResiduals.html">continuous_residuals_hook()</a></code> <code><a href="collectResiduals.html">combined_residuals_hook()</a></code> <code><a href="collectResiduals.html">deviance_residuals_hook()</a></code> <code><a href="collectResiduals.html">fitted_phat()</a></code> <code><a href="collectResiduals.html">partialScore()</a></code> </p>
        </td>
        <td><p>Residual hooks and collection methods</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="computeEtFromCt.html">computeEtFromCt()</a></code> </p>
        </td>
        <td><p>Compute the Et from the Ct</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convertMASTClassicToSingleCellAssay.html">convertMASTClassicToSingleCellAssay()</a></code> </p>
        </td>
        <td><p>Convert a MASTClassic SingleCellAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="defaultAssay.html">magic_assay_names()</a></code> <code><a href="defaultAssay.html">assay(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> </p>
        </td>
        <td><p>Default <code>assay</code> returned</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="defaultPrior.html">defaultPrior()</a></code> </p>
        </td>
        <td><p>Initialize a prior to be used a prior for BayeGLMlike/BayesGLMlike2</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dof.html">dof()</a></code> </p>
        </td>
        <td><p>Degrees of freedom of Zero inflated model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ebayes.html">ebayes()</a></code> </p>
        </td>
        <td><p>Estimate hyperparameters for hierarchical variance model for continuous component</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="expavg.html">expavg()</a></code> </p>
        </td>
        <td><p>Exponential average</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fData-methods.html">fData</a></code> </p>
        </td>
        <td><p>fData</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="featureData-methods.html">featureData</a></code> </p>
        </td>
        <td><p>Accessor for featureData <code>AnnotatedDataFrame</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterLowExpressedGenes.html">filterLowExpressedGenes()</a></code> </p>
        </td>
        <td><p>Filter low-expressing genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fit.html">fit()</a></code> </p>
        </td>
        <td><p>fit a zero-inflated regression</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="freq.html">freq()</a></code> <code><a href="freq.html">condmean()</a></code> <code><a href="freq.html">condSd()</a></code> <code><a href="freq.html">numexp()</a></code> </p>
        </td>
        <td><p>Summary statistics for genes in an experiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getConcordance.html">getConcordance()</a></code> <code><a href="getConcordance.html">getwss()</a></code> <code><a href="getConcordance.html">getss()</a></code> <code><a href="getConcordance.html">getrc()</a></code> </p>
        </td>
        <td><p>Get the concordance between two experiments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getwellKey.html">getwellKey()</a></code> </p>
        </td>
        <td><p>Accessor for wellKey</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gseaAfterBoot.html">gseaAfterBoot()</a></code> <code><a href="gseaAfterBoot.html">gsea_control()</a></code> </p>
        </td>
        <td><p>Gene set analysis for hurdle model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hushWarning.html">hushWarning()</a></code> </p>
        </td>
        <td><p>Selectively muffle warnings based on output</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="impute.html">impute()</a></code> </p>
        </td>
        <td><p>impute missing continuous expression for plotting</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="influence.bayesglm.html">influence(<i>&lt;bayesglm&gt;</i>)</a></code> </p>
        </td>
        <td><p>Influence bayesglm object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="invlogit.html">invlogit()</a></code> </p>
        </td>
        <td><p>Inverse of logistic transformation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="logFC.html">logFC()</a></code> <code><a href="logFC.html">getLogFC()</a></code> </p>
        </td>
        <td><p>Calculate log-fold changes from hurdle model components</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="logmean.html">logmean()</a></code> </p>
        </td>
        <td><p>Log mean</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lrTest-ZlmFit-character-method.html">lrTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;character&gt;</i>)</a></code> </p>
        </td>
        <td><p>Likelihood ratio test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="lrTest.html">lrTest()</a></code> </p>
        </td>
        <td><p>Run a likelihood-ratio test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="maits-dataset.html">maits</a></code> </p>
        </td>
        <td><p>MAITs data set, RNASeq</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mast_filter.html">mast_filter()</a></code> <code><a href="mast_filter.html">burdenOfFiltering()</a></code> </p>
        </td>
        <td><p>Filter a SingleCellAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="melt.SingleCellAssay.html">melt(<i>&lt;SingleCellAssay&gt;</i>)</a></code> </p>
        </td>
        <td><p>"Melt" a <code>SingleCellAssay</code> matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="model.matrix-set.html">`model.matrix<-`()</a></code> </p>
        </td>
        <td><p>Replace model matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="model.matrix.html">model.matrix()</a></code> </p>
        </td>
        <td><p>Model matrix accessor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="myBiplot.html">myBiplot()</a></code> </p>
        </td>
        <td><p>Makes a nice BiPlot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbootVcov1.html">pbootVcov1()</a></code> </p>
        </td>
        <td><p>Bootstrap a zlmfit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.thresholdSCRNACountMatrix.html">plot(<i>&lt;thresholdSCRNACountMatrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot cutpoints and densities for thresholding</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCAConcordance.html">plotSCAConcordance()</a></code> </p>
        </td>
        <td><p>Concordance plots of filtered single vs n-cell assays</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotlrt.html">plotlrt()</a></code> </p>
        </td>
        <td><p>Plot a likelihood ratio test object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="predict.ZlmFit.html">predict(<i>&lt;ZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>Return predictions from a ZlmFit object.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="predicted_sig-dataset.html">predicted_sig</a></code> </p>
        </td>
        <td><p>Predicted signatures</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="primerAverage.html">primerAverage()</a></code> </p>
        </td>
        <td><p>Average expression values for duplicated/redundant genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="print.summaryZlmFit.html">print(<i>&lt;summaryZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>Print summary of a ZlmFit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read.fluidigm.html">read.fluidigm()</a></code> </p>
        </td>
        <td><p>Reads a Fluidigm Biomark (c. 2011) raw data file (or set of files)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeResponse.html">removeResponse()</a></code> </p>
        </td>
        <td><p>Remove the left hand side (response) from a formula</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rstandard.bayesglm.html">rstandard(<i>&lt;bayesglm&gt;</i>)</a></code> </p>
        </td>
        <td><p>rstandard for bayesglm objects.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="se.coef.html">se.coef()</a></code> </p>
        </td>
        <td><p>Return coefficient standard errors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="show.html">show(<i>&lt;LMlike&gt;</i>)</a></code> <code><a href="show.html">show(<i>&lt;ZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>show</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="split-SingleCellAssay-character-method.html">split(<i>&lt;SingleCellAssay&gt;</i>,<i>&lt;character&gt;</i>)</a></code> </p>
        </td>
        <td><p>Split into <code>list</code></p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="stat_ell.html">stat_ell()</a></code> </p>
        </td>
        <td><p>Plot confidence ellipse in 2D</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subset-SingleCellAssay-method.html">subset(<i>&lt;SingleCellAssay&gt;</i>)</a></code> </p>
        </td>
        <td><p>Subset a <code>SingleCellAssay</code> by cells (columns)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summarize.html">summarize()</a></code> </p>
        </td>
        <td><p>Return programmatically useful summary of a fit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary-GSEATests-method.html">summary(<i>&lt;GSEATests&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize gene set enrichment tests</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary-ZlmFit-method.html">summary(<i>&lt;ZlmFit&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize model features from a <code>ZlmFit</code> object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary.thresholdSCRNACountMatrix.html">summary(<i>&lt;thresholdSCRNACountMatrix&gt;</i>)</a></code> <code><a href="summary.thresholdSCRNACountMatrix.html">print(<i>&lt;summaryThresholdSCRNA&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize the effect of thresholding</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="thresholdSCRNACountMatrix.html">thresholdSCRNACountMatrix()</a></code> </p>
        </td>
        <td><p>Threshold a count matrix using an adaptive threshold.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="vbeta-dataset.html">vbeta</a></code> </p>
        </td>
        <td><p>Vbeta Data Set</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="vbetaFA-dataset.html">vbetaFA</a></code> </p>
        </td>
        <td><p>Vbeta Data Set, FluidigmAssay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="waldTest-ZlmFit-matrix-method.html">waldTest(<i>&lt;ZlmFit&gt;</i>,<i>&lt;matrix&gt;</i>)</a></code> </p>
        </td>
        <td><p>Wald test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="waldTest.html">waldTest()</a></code> </p>
        </td>
        <td><p>Run a Wald test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="xform.html">xform()</a></code> </p>
        </td>
        <td><p>Make matrix of continuous expression values, orthogonal to discrete</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="zlm.html">zlm()</a></code> </p>
        </td>
        <td><p>Zero-inflated regression for SingleCellAssay</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-gene-set-enrichment-testing" class="hasAnchor"><a href="#section-gene-set-enrichment-testing" class="anchor"></a>Gene set enrichment testing</h2>
          <p class="section-desc"><p>Tests on the average differential expression effect in a gene set, accounting for gene-gene correlations. Uses bootstraps to assess the gene-gene correlations.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="gseaAfterBoot.html">gseaAfterBoot()</a></code> <code><a href="gseaAfterBoot.html">gsea_control()</a></code> </p>
        </td>
        <td><p>Gene set analysis for hurdle model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bootVcov1.html">bootVcov1()</a></code> </p>
        </td>
        <td><p>Bootstrap a zlmfit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbootVcov1.html">pbootVcov1()</a></code> </p>
        </td>
        <td><p>Bootstrap a zlmfit</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary-GSEATests-method.html">summary(<i>&lt;GSEATests&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summarize gene set enrichment tests</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-plotting-methods" class="hasAnchor"><a href="#section-plotting-methods" class="anchor"></a>Plotting methods</h2>
          <p class="section-desc"><p>Miscellaneous plotting methods. Concordance, PCA biplots, and effects plots for fitted hurdle models.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="myBiplot.html">myBiplot()</a></code> </p>
        </td>
        <td><p>Makes a nice BiPlot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotSCAConcordance.html">plotSCAConcordance()</a></code> </p>
        </td>
        <td><p>Concordance plots of filtered single vs n-cell assays</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="stat_ell.html">stat_ell()</a></code> </p>
        </td>
        <td><p>Plot confidence ellipse in 2D</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-loading-transforming-and-summarizing">Loading, transforming and summarizing</a></li>
      <li><a href="#section-thresholding">Thresholding</a></li>
      <li><a href="#section-differential-expression-testing-with-hurdle-model">Differential Expression Testing with Hurdle Model</a></li>
      <li><a href="#section-specifying-hypothesis-and-contrasts-to-test">Specifying hypothesis and contrasts to test</a></li>
      <li><a href="#section-low-level-manipulation-of-fitted-hurdle-models">Low-level manipulation of fitted hurdle models</a></li>
      <li><a href="#section-gene-set-enrichment-testing">Gene set enrichment testing</a></li>
      <li><a href="#section-plotting-methods">Plotting methods</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

