<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Calculate log-fold changes from hurdle model components — logFC • MAST</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Calculate log-fold changes from hurdle model components — logFC" />

<meta property="og:description" content="Using the delta method, estimate the log-fold change from a state given by a vector contrast0 and the state(s) given by contrast1." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/MAITAnalysis.html">Using MAST with RNASeq: MAIT Analysis.</a>
    </li>
    <li>
      <a href="../articles/MAST-Intro.html">MAST Intro</a>
    </li>
    <li>
      <a href="../articles/MAST-interoperability.html">Interoptability between MAST and SingleCellExperiment-derived packages.</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/RGLab/MAST/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate log-fold changes from hurdle model components</h1>
    <small class="dont-index">Source: <a href='https://github.com/RGLab/MAST//blob/master/R/ZlmFit-logFC.R'><code>R/ZlmFit-logFC.R</code></a></small>
    <div class="hidden name"><code>logFC.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Using the delta method, estimate the log-fold change from a state given by a vector contrast0 and the state(s) given by contrast1.</p>
    
    </div>

    <pre class="usage"><span class='fu'>logFC</span>(<span class='no'>zlmfit</span>, <span class='no'>contrast0</span>, <span class='no'>contrast1</span>)

<span class='fu'>getLogFC</span>(<span class='no'>zlmfit</span>, <span class='no'>contrast0</span>, <span class='no'>contrast1</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>zlmfit</th>
      <td><p>ZlmFit output</p></td>
    </tr>
    <tr>
      <th>contrast0</th>
      <td><p>vector of coefficients giving baseline contrast, or a <code><a href='Hypothesis.html'>Hypothesis</a></code>.  If missing, then the '(Intercept)' is used as baseline.</p></td>
    </tr>
    <tr>
      <th>contrast1</th>
      <td><p>matrix of coefficients giving comparison contrasts, or a <code><a href='Hypothesis.html'>Hypothesis</a></code>.  If missing, then all non-(Intercept) coefficients are compared.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list of matrices `logFC` and `varLogFC`, giving the log-fold-changes for each contrast (columns) and genes (rows) and the estimated sampling variance thereof</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The log-fold change is defined as follows.  For each gene, let \(u(x)\) be the expected value of the continuous component, given a covariate x and the estimated coefficients <code>coefC</code>, ie, \(u(x)=\) <code>crossprod(x, coefC)</code>.
Likewise, Let \(v(x)=\) <code>1/(1+exp(-crossprod(coefD, x)))</code> be the expected value of the discrete component.
 The log fold change from contrast0 to contrast1 is defined as
$$u(contrast1)v(contrast1)-u(contrast0)v(contrast0).$$
Note that for this to be a log-fold change, then the regression for u must have been fit on the log scale.  This is returned in the matrix <code>logFC</code>.
An approximation of the variance of <code>logFC</code> (applying the delta method to  formula defined above) is provided in <code>varLogFC</code>.</p>
    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>getLogFC</code>: Return results as a perhaps friendlier <code>data.table</code></p></li>
</ul>
    
    <h2 class="hasAnchor" id="caveats"><a class="anchor" href="#caveats"></a>Caveats</h2>

    
    <p>1.  When <code>method='bayesglm'</code> (the default), it's no longer necessarily true that the log fold change from condition A to B will be the inverse of the log fold change from B to A if the models are fit separately.
This is due to the shrinkage in <code>bayesglm</code>.</p>
<p>2.  The log fold change can be small, but the Hurdle p-value small and significant when the sign of the discrete and continuous model components are discordant 
so that the marginal log fold change cancels out.
The large sample sizes present in many single cell experiments also means that there is substantial power to detect even small changes.</p>
<p>3.  When there is no expression in a gene for a coefficient that is non-zero in either <code>condition0</code>
or <code>condition1</code> we return <code>NA</code> because there is not any information
to estimate the continuous component.  Technically we might return plus or minus infinity,
but there is not a straightforward way to estimate a confidence interval in any case.
See <a href='https://support.bioconductor.org/p/99244/'>https://support.bioconductor.org/p/99244/</a> for details</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='Hypothesis.html'>Hypothesis</a></p>
<p><a href='summary-ZlmFit-method.html'>summary,ZlmFit-method</a></p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>vbetaFA</span>)
<span class='no'>zz</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='zlm.html'>zlm</a></span>( ~ <span class='no'>Stim.Condition</span>+<span class='no'>Population</span>, <span class='no'>vbetaFA</span>[<span class='fl'>1</span>:<span class='fl'>5</span>,])</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Done!</span></div><div class='input'><span class='co'>##log-fold changes in terms of intercept (which is Stim(SEB) and CD154+VbetaResponsive)</span>
<span class='no'>lfcStim</span> <span class='kw'>&lt;-</span> <span class='fu'>logFC</span>(<span class='no'>zz</span>)
<span class='co'>##If we want to compare against unstim, we can try the following</span>
<span class='no'>coefnames</span> <span class='kw'>&lt;-</span> <span class='fu'>colnames</span>(<span class='fu'>coef</span>(<span class='no'>zz</span>, <span class='st'>'D'</span>))
<span class='no'>contrast0</span> <span class='kw'>&lt;-</span> <span class='fu'>setNames</span>(<span class='fu'>rep</span>(<span class='fl'>0</span>, <span class='fu'>length</span>(<span class='no'>coefnames</span>)), <span class='no'>coefnames</span>)
<span class='no'>contrast0</span>[<span class='fu'>c</span>(<span class='st'>'(Intercept)'</span>, <span class='st'>'Stim.ConditionUnstim'</span>)] <span class='kw'>&lt;-</span> <span class='fl'>1</span>
<span class='no'>contrast1</span> <span class='kw'>&lt;-</span> <span class='fu'>diag</span>(<span class='fu'>length</span>(<span class='no'>coefnames</span>))
<span class='fu'>rownames</span>(<span class='no'>contrast1</span>)<span class='kw'>&lt;-</span><span class='fu'>colnames</span>(<span class='no'>contrast1</span>)<span class='kw'>&lt;-</span><span class='no'>coefnames</span>
<span class='no'>contrast1</span>[<span class='st'>'(Intercept)'</span>,]<span class='kw'>&lt;-</span><span class='fl'>1</span>
<span class='no'>lfcUnstim</span> <span class='kw'>&lt;-</span> <span class='fu'>logFC</span>(<span class='no'>zz</span>, <span class='no'>contrast0</span>, <span class='no'>contrast1</span>)
<span class='co'>##log-fold change with itself is 0</span>
<span class='fu'>stopifnot</span>(<span class='fu'>all</span>(<span class='no'>lfcUnstim</span>$<span class='no'>logFC</span>[,<span class='fl'>2</span>]<span class='kw'>==</span><span class='fl'>0</span>))
<span class='co'>##inverse of log-fold change with Stim as reference</span>
<span class='fu'>stopifnot</span>(<span class='fu'>all</span>(<span class='no'>lfcStim</span>$<span class='no'>logFC</span>[,<span class='fl'>1</span>]<span class='kw'>==</span>(-<span class='no'>lfcUnstim</span>$<span class='no'>logFC</span>[,<span class='fl'>1</span>])))
<span class='co'>##As a data.table:</span>
<span class='fu'>getLogFC</span>(<span class='no'>zz</span>)</div><div class='output co'>#&gt;     primerid                          contrast       logFC   varLogFC
#&gt;  1:   B3GAT1              Stim.ConditionUnstim  0.02121690 0.05699408
#&gt;  2:   B3GAT1 PopulationCD154-VbetaUnresponsive  0.09823225 0.11759250
#&gt;  3:   B3GAT1   PopulationCD154+VbetaResponsive  0.97402709 0.26295565
#&gt;  4:   B3GAT1 PopulationCD154+VbetaUnresponsive  0.50171226 0.33509425
#&gt;  5:   B3GAT1         PopulationVbetaResponsive  0.07638735 0.12164060
#&gt;  6:   B3GAT1       PopulationVbetaUnresponsive         NaN        NaN
#&gt;  7:      BAX              Stim.ConditionUnstim -0.42956548 1.35480633
#&gt;  8:      BAX PopulationCD154-VbetaUnresponsive -1.32492734 2.58024639
#&gt;  9:      BAX   PopulationCD154+VbetaResponsive  0.62690532 2.65894498
#&gt; 10:      BAX PopulationCD154+VbetaUnresponsive -0.66503358 3.49255200
#&gt; 11:      BAX         PopulationVbetaResponsive  1.70757502 3.10585739
#&gt; 12:      BAX       PopulationVbetaUnresponsive  1.95779243 3.15029439
#&gt; 13:     BCL2              Stim.ConditionUnstim -0.45315003 0.10064027
#&gt; 14:     BCL2 PopulationCD154-VbetaUnresponsive  1.47536864 0.56464034
#&gt; 15:     BCL2   PopulationCD154+VbetaResponsive  1.70251880 0.57023184
#&gt; 16:     BCL2 PopulationCD154+VbetaUnresponsive  2.22719080 1.15781350
#&gt; 17:     BCL2         PopulationVbetaResponsive  3.13232752 1.14348444
#&gt; 18:     BCL2       PopulationVbetaUnresponsive         NaN        NaN
#&gt; 19:     CCL2              Stim.ConditionUnstim  0.80636176 1.56877955
#&gt; 20:     CCL2 PopulationCD154-VbetaUnresponsive  0.42039146 0.53040577
#&gt; 21:     CCL2   PopulationCD154+VbetaResponsive  0.68379249 0.54689602
#&gt; 22:     CCL2 PopulationCD154+VbetaUnresponsive  0.64470476 0.88561851
#&gt; 23:     CCL2         PopulationVbetaResponsive -0.41102065 0.32445543
#&gt; 24:     CCL2       PopulationVbetaUnresponsive -0.42352278 0.32101503
#&gt; 25:     CCL3              Stim.ConditionUnstim -0.96766753 0.43167759
#&gt; 26:     CCL3 PopulationCD154-VbetaUnresponsive  0.48902668 1.10452054
#&gt; 27:     CCL3   PopulationCD154+VbetaResponsive  3.17114539 1.49031333
#&gt; 28:     CCL3 PopulationCD154+VbetaUnresponsive  1.83445067 2.37771601
#&gt; 29:     CCL3         PopulationVbetaResponsive  0.32428970 1.31906307
#&gt; 30:     CCL3       PopulationVbetaUnresponsive -0.14312747 1.11905570
#&gt;     primerid                          contrast       logFC   varLogFC
#&gt;               z
#&gt;  1:  0.08887244
#&gt;  2:  0.28646019
#&gt;  3:  1.89945839
#&gt;  4:  0.86670485
#&gt;  5:  0.21901918
#&gt;  6:         NaN
#&gt;  7: -0.36905472
#&gt;  8: -0.82482435
#&gt;  9:  0.38445656
#&gt; 10: -0.35585423
#&gt; 11:  0.96892249
#&gt; 12:  1.10303953
#&gt; 13: -1.42842066
#&gt; 14:  1.96342626
#&gt; 15:  2.25458278
#&gt; 16:  2.06984640
#&gt; 17:  2.92922030
#&gt; 18:         NaN
#&gt; 19:  0.64379702
#&gt; 20:  0.57723115
#&gt; 21:  0.92463844
#&gt; 22:  0.68507408
#&gt; 23: -0.72158271
#&gt; 24: -0.74750498
#&gt; 25: -1.47280896
#&gt; 26:  0.46531350
#&gt; 27:  2.59763042
#&gt; 28:  1.18966873
#&gt; 29:  0.28235818
#&gt; 30: -0.13529979
#&gt;               z</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#functions">Functions</a></li>

      <li><a href="#caveats">Caveats</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Andrew McDavid, Greg Finak, Masanao Yajima.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

